processes/releasing.md

```diff
@@ -0,0 +1,85 @@
+1. [Release IPAM](https://github.com/metal3-io/ip-address-manager/blob/main/docs/releasing.md)
+1. Uplift BMO to use latest go modules as in CAPI (controller runtime,
+   `k8s.io/*` in `go.mod`)
+1. [Release BMO](https://github.com/metal3-io/baremetal-operator/blob/main/docs/releasing.md)
+1. Uplift CAPM3 to use latest CAPI, IPAM, BMO (in `go.mod`)
+1. [Release CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/docs/releasing.md)
+1. Update Metal3 dev-env and CI to support new releases
+
+## Update Metal3 dev-env and CI to support new releases
+
+If a new minor or major was released, we need to make changes in Metal3 dev-env
+and CI.
+
+### Tag Ironic and Mariadb images
+
+After releasing v1.x.y version of CAPM3, create an annotated tag with `capm3-`
+prefix + release version in
+[Ironic-image](https://github.com/metal3-io/ironic-image) and
+[mariadb-image](https://github.com/metal3-io/mariadb-image) Github
+repositories.
+
+```bash
+git tag -s -a capm3-v1.x.y -m capm3-v1.x.y [optional sha, or tag^{}]
+git push origin capm3-v1.x.y
+```
+
+After this, [Ironic](https://quay.io/repository/metal3-io/ironic) and
+[Mariadb](https://quay.io/repository/metal3-io/mariadb) container images will
+be built automatically  with `capm3-v1.x.y` tag in Quay. Verify the build is
+triggered, as often Quay has disabled the build trigger due build failures or
+other reasons.
+
+### Create Jenkins jobs
+
+Two Jenkins jobs need to be created in
+[JJB](https://gerrit.nordix.org/plugins/gitiles/infra/cicd/+/refs/heads/master/jjb/metal3/):
+
+- a periodic job that runs on a regular basis.
+- a PR verification job that is triggered by a keyword on a PR targeted for that
+  release branch.
+
+We also need to change project-infra keywords for required tests for some of
+the repositories.
+
+### Dev-env changes
+
+Metal3-dev-env variables need to be modified. After a major or minor release,
+the new minor version should point to main for CAPM3/IPAM and the released
+version should point to the release branch.
+
+WARNING: this breaks metal3-dev-env (CI included) until dev-env and IPAM/CAPM3
+repositories are in sync.
+
+### Update testing matrix documentation
+
+Update [testing matrix documentation](https://github.com/metal3-io/metal3-docs/pull/320)
+to reflect the new release and the combination of components that are tested in
+periodic jobs.
+
+TODO: change PR link to document link after PR is merged.
+
+## BMO releasing to be discussed
+
+1. BMO releasing and maintenance
+   - BMO does not have release branches, only tags off the `main` branch
+   - Without release branches, maintaining BMO versions that are coupled with CAPM3 releases is impossible
+1. BMO contains multiple subcomponents and images:
+   - keepalived ([within BMO repo](https://github.com/metal3-io/baremetal-operator/tree/main/resources/keepalived-docker))
+   - [mariadb-image](https://github.com/metal3-io/mariadb-image)
+   - [ironic-image](https://github.com/metal3-io/ironic-image)
```

Comment:
BMO does not contain these two, maybe change the wording?

Reply:
This is WIP. The wording will be changed quite a bit, especially everything around BMO.

Reply:
Done. Let me know if the wording is OK, and if not, please propose a better wording.

Reply:
I'm not sure what you're intending to say. Maybe "BMO makes strong assumption about the exact configuration of Ironic. However, [ironic-image](https://github.com/metal3-io/ironic-image) is currently branchless."

Reply:
Thanks. Used your wording and cut down a lot of the details, they were unnecessary here.